pkg/networkservice/core/next/tail_server.go

```diff
@@ -24,10 +24,15 @@ import (
 	"github.com/networkservicemesh/api/pkg/api/networkservice"
 )
 
-// tailServer is a simple implementation of networkservice.NetworkServiceServer that is called at the end of a chain
-// to insure that we never call a method on a nil object
 type tailServer struct{}
 
+// TailServer is a simple implementation of networkservice.NetworkServiceServer that is:
+// - called at the end of a chain to insure that we never call a method on a nil object
+// - used to break an existing chain
+func TailServer() networkservice.NetworkServiceServer {
```

Comment:
Why do we need to make this public?

Reply:
It is used in `timeout` to prevent such issue:
```
... -> timeout (we are here) -> after-timeout -> ...

newCtx := extend.WithValuesFromContext(context.Background(), ctx)
time.AfterFunc(duration, func() {
    closeChain.Close(newCtx, conn)
}
```
when `time.AfterFunc` fires, we got:
```
next.Server(newCtx) == after-timeout
```
so the close chain is not:
```
closeChain
```
but:
```
closeChain -> after-timeout -> ...
```
If we use `next.TailServer()` we got:
```
closeChain -> next.TailServer() -> after-timeout -> ...
```
and it is just the same as:
```
closeChain
```

Moreover, there is one more use case in cmd-forwarder-sriov (dummyServer == next.TailServer()): https://github.com/networkservicemesh/cmd-forwarder-sriov/pull/18/files#diff-cabad8ab567c8dcae02dfa95a8704fda0fd1753970eb70ddc71fec364618a58f
A wrapper chain element to close only wrapped chain elements, but not all subsequent chain.

Reply:
Can we rollback tails element back and create other ones for `timeout` goals?

Reply:
Replaced with `chainbreak`.